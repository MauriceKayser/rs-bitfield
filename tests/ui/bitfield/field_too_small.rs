//! Assertion failure: _TYPE_IN_FIELD_0_IS_SMALLER_THAN_THE_SPECIFIED_SIZE_OF_9_BITS

extern crate alloc;

#[bitfield::bitfield(16)]
struct BitField(#[field(0, 9)] Field); // `Field` is only 8 bits wide, and can not store 9 bits.

#[derive(Debug, bitfield::Field)]
#[repr(u8)]
enum Field {
    F1 = 1,
    F2,
    F3
}

fn main() {}