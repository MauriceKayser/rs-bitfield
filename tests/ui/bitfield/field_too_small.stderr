error[E0080]: evaluation of constant value failed
 --> tests/ui/bitfield/field_too_small.rs:6:32
  |
6 | struct BitField(#[field(0, 9)] Field); // `Field` is only 8 bits wide, and can not store 9 bits.
  |                                ^^^^^ the evaluated program panicked at 'Type is smaller than the specified size of 9 bits', $DIR/tests/ui/bitfield/field_too_small.rs:6:32
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: the variant or associated item `_bitfield_extract_primitive` exists for enum `Field`, but its trait bounds were not satisfied
  --> tests/ui/bitfield/field_too_small.rs:6:32
   |
6  | struct BitField(#[field(0, 9)] Field); // `Field` is only 8 bits wide, and can not store 9 bits.
   |                                ^^^^^ variant or associated item cannot be called on `Field` due to unsatisfied trait bounds
...
10 | enum Field {
   | ---------- doesn't satisfy 6 bounds
   |
note: the following trait bounds were not satisfied:
      `&Field: TryFrom<i16>`
      `&Field: TryFrom<u16>`
      `&mut Field: TryFrom<i16>`
      `&mut Field: TryFrom<u16>`
      `<&Field as TryFrom<i16>>::Error = i16`
      `<&Field as TryFrom<u16>>::Error = u16`
      `<&mut Field as TryFrom<i16>>::Error = i16`
      `<&mut Field as TryFrom<u16>>::Error = u16`
      `<Field as TryFrom<i16>>::Error = i16`
      `<Field as TryFrom<u16>>::Error = u16`
      `Field: TryFrom<i16>`
      `Field: TryFrom<u16>`
  --> tests/ui/bitfield/field_too_small.rs:6:28
   |
6  | struct BitField(#[field(0, 9)] Field); // `Field` is only 8 bits wide, and can not store 9 bits.
   |                            ^^^^^^^^^
   |                            |
   |                            unsatisfied trait bound introduced here
note: the trait `TryFrom` must be implemented
  --> /builddir/build/BUILD/rustc-1.86.0-src/library/core/src/convert/mod.rs:681:1
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `_bitfield_extract_primitive`, perhaps you need to implement one of them:
           candidate #1: `BitFieldExtractionSigned`
           candidate #2: `BitFieldExtractionUnsigned`
